text/0089-metrics-views-api.md

```diff
@@ -0,0 +1,264 @@
+It proposes a specification for a _view API_ that allows application owners and administrators to configure aggregations for individual metric instruments.
+
+## TL;DR
+
+The potentially contentious changes this OTEP proposes include:
+
+- Require views to be registered: don't record measurements from API for metric instruments that don't appear in any view.
+- Drop support for LastValue aggregations, don't allow multiple Observer calls per export interval.
+- Don't allow views to specify how "dropped" keys are aggregated, since we can safely ignore aggregation order for measurement instruments other than observers.
+
+## Motivation
+
+As of [opentelemetry-specification#430](https://github.com/open-telemetry/opentelemetry-specification/pull/430), the spec describes three different metric instrument types, and a default aggregation for each.
+These default aggregations are consistent with the semantics of each metric instrument, but not with the various exposition formats required by APM backends.
+
+For example, consider the [Prometheus histogram exposition format](https://prometheus.io/docs/concepts/metric_types/#histogram).
+The default aggregation for the Measure metric instrument is _MinMaxSumCount_.
+This aggregation computes summary statistics about (i.e. the _min_, _max_, _sum_, and _count_ of) the set of recorded values, but doesn't capture enough information to reconstruct histogram bucket counts.
+
+If instead the Measure instrument used an [_Exact_ aggregation](https://github.com/open-telemetry/opentelemetry-specification/pull/347/files#diff-5b01bbf3430dde7fc5789b5919d03001R254-R259) -- which stores the sequence of measurements without aggregating them -- the Prometheus exporter could compute the histogram for each recording interval at export time, but at the cost of significantly increased memory usage.
+
+A better solution would be to use a custom _Histogram_ aggregation, which could track bucket counts and compute summary statistics without preserving the original measurements.
+
+As the spec is written now, all metrics are aggregated following the default aggregation for the metric instrument by which they were recorded.
+By design, there is no option to change these defaults:
+
+> [Users do not have a facility in the API to select the aggregation they want for particular instruments.](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics.md#aggregations)
+
+This limitation prevents users from:
+
+- Converting aggregated data to vendor-specific exposition formats
+- Configuring aggregation options, e.g. histogram bucket boundaries
+- Specifying multiple aggregations for a single metric
+
+In addition, it's often desirable to aggregate metrics with respect to a specific set of label keys to reduce memory use and export data size.
```

Comment:
Would it be correct to say that Views is the only way to pick/drop dimensions? Or Views is just one way. And batcher can still drop dimensions further?